### PR TITLE
feat: Schema Update

### LINKED ISSUE #18: feat: Schema update
### Describe the feature you'd like to request

As a user I want to be able to write reviews, leave ratings, how much I spent, and the date I went somewhere.

### Describe the solution you'd like to see

A new schema to keep track of: restaurantName, reviewDescription, stars, price, date, user, and displayName.

### Additional information

The old schema needs to be updated.

### PR DESCRIPTION
**Pull request by Nebula5102**
Beginning to update schema to use the following structure:
restaurantName -> str
reviewDescription -> str
stars -> int
price -> float 2 decimals
date -> Date
user -> Discord username
displayName -> str

Resolves #18

### CODE REVIEW

**Comment by Nebula5102** [comment]
Updated database schema. Started to make changes to posts and form for making posts, but did not finish any features. Create post button on review page is currently not working. The new schema looks like this:
```
  id: int("id").autoincrement().primaryKey(),
  restaurantName: varchar("restaurantName", { length: 256 }),
  reviewText: varchar("reviewText", { length: 256 }),
  stars: int("stars").default(5),
  price: float("price"),
  date: datetime("date"),
  displayName: text("displayName"),
  createdAt: timestamp("created_at")
    .default(sql`CURRENT_TIMESTAMP`)
    .notNull(),
  updatedAt: timestamp("updatedAt").onUpdateNow(),
  ```

**Comment by Nebula5102** [comment]
Schema is fully updated and usable. Ready to be merged to main for milestone 1. Currently we are missing the functionality of a authorized users being able to make a post and have the username field in the schema be updated to that users username. We will need to fix this in the future for sorting of posts by user etc. Immediate todos: 
Update styling of review page
implement a way to update fields of post using user input
Update styling of feed page
Update styling of posts

**Comment by callumcurtis** [comment]
Resolves #18

**Review by callumcurtis — CHANGES_REQUESTED** [review]
I think it's important to fix the UI issues for this milestone 1 but the security and schema issues can be resolved after M1, included them in the review so that someone can refer to them later when working on this code later

File: apps/nextjs/src/app/_components/posts.tsx [hunk 1]
```diff
@@ -92,14 +134,14 @@ export function PostList(props: {
...
-        <div className="absolute inset-0 flex flex-col items-center justify-center bg-black/10">
-          <p className="text-2xl font-bold text-white">No posts yet</p>
+      <div className="relative flex w-full flex-col content-center gap-4">
+        <div className="mt-20 max-w-xs">
+          <p className="text-base text-black">
+            Your feed is looking a bit <span className="italic">hungry...</span>{" "}
+            Maybe it is time to find a place to eat!
+          </p>
         </div>
+        <div className=""></div>
```
[the remark this answers is not in the document]
Comment (Nebula5102) [reply apps/nextjs/src/app/_components/posts.tsx:144]: removed, it is not needed

File: apps/nextjs/src/app/_components/posts.tsx [hunk 2]
```diff
@@ -92,14 +134,14 @@ export function PostList(props: {
...
-      <div className="relative flex w-full flex-col gap-4">
-        <PostCardSkeleton pulse={false} />
-        <PostCardSkeleton pulse={false} />
-        <PostCardSkeleton pulse={false} />
-
-        <div className="absolute inset-0 flex flex-col items-center justify-center bg-black/10">
-          <p className="text-2xl font-bold text-white">No posts yet</p>
+      <div className="relative flex w-full flex-col content-center gap-4">
+        <div className="mt-20 max-w-xs">
+          <p className="text-base text-black">
```
[the remark this answers is not in the document]
Comment (Nebula5102) [reply apps/nextjs/src/app/_components/posts.tsx:139]: implemented suggestion

File: apps/nextjs/src/app/_components/posts.tsx [hunk 3]
```diff
@@ -92,14 +134,14 @@ export function PostList(props: {
...
   if (posts.length === 0) {
     return (
-      <div className="relative flex w-full flex-col gap-4">
-        <PostCardSkeleton pulse={false} />
-        <PostCardSkeleton pulse={false} />
-        <PostCardSkeleton pulse={false} />
-
-        <div className="absolute inset-0 flex flex-col items-center justify-center bg-black/10">
-          <p className="text-2xl font-bold text-white">No posts yet</p>
+      <div className="relative flex w-full flex-col content-center gap-4">
```
[the remark this answers is not in the document]
Comment (Nebula5102) [reply apps/nextjs/src/app/_components/posts.tsx:137]: implemented suggestion

**Pull request closed without merging**